src/ui/handlers/starter-select-ui-handler.ts

```diff
@@ -2266,95 +2265,77 @@ export class StarterSelectUiHandler extends MessageUiHandler {
    * @param props the current {@linkcode DexAttrProps} for the selected Pokemon
    * @returns `true` if the 'success' sound effect should be played
    */
-  handleCycleShiny(starterPrefs: StarterAttributes, props: DexAttrProps): boolean {
-    if ((!props.shiny && this.canToggleShiny) || (props.shiny && !this.canCycleVariant)) {
-      return this.toggleShinyState(starterPrefs);
+  private handleCycleShiny(starterPrefs: StarterAttributes, props: DexAttrProps): boolean {
+    if (isNil(this.speciesStarterDexEntry)) {
+      return false;
     }
-    if (props.shiny && this.canCycleVariant) {
-      // Find next unlocked variant
-      const previousVariant = isNil(starterPrefs.variant) ? props.variant : starterPrefs.variant;
-      let variant = previousVariant;
-      do {
-        variant = (variant + 1) % 3;
-        if (variant === 0 && this.canToggleShiny) {
-          // If we cycled back to the first variant and have the non shiny form unlocked
-          // we disable shiny state instead of looking through the other variants
-          return this.toggleShinyState(starterPrefs);
-        }
-        if (!isNil(this.speciesStarterDexEntry)) {
-          if (variant === 0 && this.speciesStarterDexEntry.caughtAttr & DexAttr.DEFAULT_VARIANT) {
-            return this.switchToVariant(starterPrefs, variant);
-          }
-          if (variant === 1 && this.speciesStarterDexEntry.caughtAttr & DexAttr.VARIANT_2) {
-            return this.switchToVariant(starterPrefs, variant);
-          }
-          if (variant === 2 && this.speciesStarterDexEntry.caughtAttr & DexAttr.VARIANT_3) {
-            return this.switchToVariant(starterPrefs, variant);
-          }
+
+    const variants = [
+      DexAttr.NON_SHINY,
+      DexAttr.SHINY_BASE_VARIANT,
+      DexAttr.SHINY_RARE_VARIANT,
+      DexAttr.SHINY_EPIC_VARIANT,
+    ];
+    const currentShinyVariant = isNil(starterPrefs.variant) ? props.variant : starterPrefs.variant;
```

Comment:
I think this could be simplified to this?
```suggestion
    const currentShinyVariant = starterPrefs.variant ?? props.variant;
```